### DISCUSSION TITLE
Concept of Driver using Pass_Through / passing information to another device in XCSoar

### DISCUSSION THREAD

**Discussion by SashLi** [XCSoar/XCSoar#844] [Q&A]
Hi XCSoar community,
are there any further information to the PASS_THROUGH option in the XCSoar drivers ?
As I understand a driver set the key that it want / can do to pass through data to another unit.

If the key is enabled the secondary device driver can be chosen in the dervice setup of the driver.

My question is where is the point the data is given to the other device. Only found the function EnablePassThrough that can be overwritten. It there another method that is trigged when the PassThrough is needed or is this handelt by XCSoar ? Does the NMEA  output is automatically piped to any other device that is using the secondary driver , that was selected in the driver setup.

Reason is I want to send a command from one NMEA device to another NMEA device with using XCSoar as center of all communication. Both devices are connected to the XCSoar via Serial and listen on incoming NEA commands. 

Now device A needs to inform device B that a status was changed that is not represended by one of the "default" NMEA sense.

Would love to here from any of you.

Regards,
Sascha

**Comment by MaxKellermann** [comment]
This has nothing to do with NMEA. It is expected that a device which has a pass-through mode will mix all the NMEA from itself and the secondary device.

Pass-through mode is use to access a device with (proprietary) non-NMEA protocols which is chained behind another device. This was developed for the LXNAV V7 vario which can be connected to a FLARM. All commands sent to the V7 go only to the V7, so you can't declare a task. But the V7 has a command to enable pass-through mode, which ceases all V7 transfer, and forwards everything as-is between XCSoar and the FLARM. In this mode, you can declare tasks and download IGC files from the FLARM.
